### ISSUE TITLE
Multiple languages support.

### ISSUE THREAD

**Issue by Rii-** [Sidetalker/SplatPal#2]
As the title says, it would greatly improve the app's public radius. But some people like me doesn't like to be forced to have the app in their language, so having an option to choose languages rather than it being determined by the system language would be great. As I said, I'm highly motivated to do the french translation. As for other languages, if you don't know where to ask there's a /r/Jailbreak page just for that ! : https://www.reddit.com/r/jailbreak/wiki/translators It is supposed to be for tweak translation but I highly doubt that anyone would refuse you just because it's an AppStore app. ;)

**Comment by Sidetalker** [comment]
I think you mentioned this early on - maybe in your initial message to me. If not, you’re the second person to mention it meaning demand is higher than I anticipated. This is something I’m keeping on my roadmap, but I intend to do a public release before diving into it. I definitely appreciate your offer and will keep it in mind when I go about adding this functionality. I think you’re right that a setting is a good way to go - customization is always welcome.

**Comment by Rii-** [comment]
Yeah that was me, that's why I said "As I said". I was just adding my suggestions to your issue tracker as you requested. ;)

**Comment by Sidetalker** [comment]
Much appreciated!

**Comment by Sidetalker** [comment]
#23 EU gear support. @Rii- if you're receiving updates on this, I'd definitely take you up on your offer to do french localization. The gear information is loaded from this file: https://github.com/Sidetalker/SplatPal/blob/master/SplatPal/Resources/data.json and French localization (for the gear at least) could be done by adding name_fr to the gear portions of the data. If you're interested in localizing the full app I would need to prepare all of my strings for localization and give you the resource, but I'm definitely willing to do that at this stage as well.

**Comment by Rii-** [comment]
Hi Kevin,

I would love to translate your entire app ! I'm going to take a look at the gear translation file in a day or two and I will let you know when I'm done so we can continue.

Have a great day !

**Comment by Sidetalker** [comment]
Awesome! I'll start preparing the rest of the app for localization. I'll plan to include gear names in on the minor release I'm hoping to get out this week and localize the rest of the app before a +0.1.0 release.

**Comment by Rii-** [comment]
I added french names to the first 3 gear, tell me if i'm doing it right so I can continue. Also, I forgot that for whatever reason video games compagnies like to make a difference between France's French and Canada's French (just like US and UK) so "Aloha shirt" is "Chemisette tropiques" for NOE and "Chemise Hawaïenne" for NOA(I don't even know why because as a french I would have said "Chemise Hawaïenne". Anyway, I can do both translations if you want, just tell me what I'm supposed to write after "name_".
[SplatPal Gear_fr.txt](https://github.com/Sidetalker/SplatPal/files/155017/SplatPal.Gear_fr.txt)

**Comment by Sidetalker** [comment]
One thing - add a comma after all the entries accept the last one. So 

>  "name":"18K Aviators"
> "name_fr":"Lunettes de soleil 18 carats"

would become

>  "name":"18K Aviators",
> "name_fr":"Lunettes de soleil 18 carats"

That's easy enough for me to change in bulk, but you may as well be doing it as you go through. As for multiple locales with the same language... here's a list of the corresponding language codes: https://msdn.microsoft.com/en-us/library/ms533052(v=vs.85).aspx. According to that list, France's French is fr-fr and Canada's is fr-ca. If you're willing to do both, go ahead and use the convention name_[languageCode]:

>  "name":"Aloha Shirt",
> "name_fr-fr":"Chemisette tropiques",
> "name_fr-ca":"Chemise Hawaïenne"

Thanks so much for doing this work, man. I've got an update submitted to the store already because I really want to get my fixes for the few crash bugs out, but it won't go through for at least 3-4 more days. If this is done before it goes live, I'll pull it and resubmit with your work included.

**Comment by Rii-** [comment]
Hurray ! I'm gonna work on it right now ! Let me know when I can work on the rest of the app ! :)

**Comment by Rii-** [comment]
Do I still need to put fr-fr and fr-ca if there is no difference ? Or will just fr-fr do the job ?

**Comment by Sidetalker** [comment]
Just fr-fr will be alright. I’ll default to fr-fr for fr-[anything] if there’s no fr-[anything] name.

**Comment by Rii-** [comment]
Hey !

I've finally finished translating the gear file, after more than one month, I know. But actually what took me so long is that I thought that since I was on the wiki for the French translations, why not add the others ? So this file contains translations in French (fr-fr), Canadian French (fr-ca), Spanish (es-es), Latin American Spanish (es-xl), German (de) and Italian (it).

[SplatPal Gear_fr+es+de+it.txt](https://github.com/Sidetalker/SplatPal/files/207510/SplatPal.Gear_fr.es.de.it.txt)

Unfortunately, there was some translations missing from the wiki so I had to change my Wii U's language a few times and figure it out myself. But since I have a PAL Wii U I don't have access to Canadian French and Latin American Spanish.

Here are the remaining missing translations :
Safari Hat : Spanish (es-xl) and French (NOA)
Slipstream United : French (NOA) was not listed so it might be different from French (NOE)
Splatfest Tee : Spanish (NOA) was not listed so it might be different from Spanish (NOE)

Maybe if you have time you could change your Wii U's language too to complete the file.

Other than that the file is complete, just make sure that you have the right encoding format for your file because French uses "é" and "è" and German uses "ß".

Just to make things clear, I am able to translate Splatoon ressources from others languages thanks to wiki but I will only be able to translate the app itself in french.

If you're happy with my work I would love to continue translating your app ! Just let me know which file is next. :)

Have a great day !

**Comment by Sidetalker** [comment]
Wow man, fantastic work. I've been slacking a bit on development and will use this to kick myself back into gear. I'll follow up in the next week or so with information on localizing the rest. I'll definitely take the time to fill in the few blanks you have... really, really good job with this man, thank you.

**Comment by Rii-** [comment]
Thanks ! I'm glad that you like my work.
Looking forward to continuing working with you !

**Comment by Rii-** [comment]
Hi !
I noticed that you updated SplatPal and I'm really excited to test it ! However, that last update requires IOS 9 and as I already mentioned I'm currently jailbroken on IOS 8.4 and I am obviously not willing to loose my jailbreak anytime soon.
What feature did you add requires IOS 9 ? I might be able to bypass the IOS 9 restriction but I wanted to know if that IOS 9 update was justified or not. :)
Thanks !

**Comment by Sidetalker** [comment]
Sorry about that, I set the minimum requirements to iOS 9 so I could try out the new UIStackView. I don't really need to use it anywhere (the one place I've used it could be done just fine with normal constraints). I'll remove that restriction and upload another build tonight. The current build also has all of the localization data you painstakingly added for me (thanks again!) - I just ran out of time and didn't write the code to actually enable it. You can expect to see your French translations in the next build I push!

**Comment by Sidetalker** [comment]
The latest build has your gear localizations along with iOS 8.4 support, thanks again for your help!

**Comment by Rii-** [comment]
Thanks but it still shows english names in the Gear section and I noticed that often, when I enter the gear section the app will crash while scrolling, sometimes right away, sometimes after 20 seconds but I can't seem to find a speficif pattern that trigger the crash...

**Comment by Sidetalker** [comment]
The next build will include an option in Settings to manually select your language. I'll also add a display showing what iOS _thinks_ your language is so we can try to debug why it's not showing up for you. That crash in gear is interesting... What iOS device do you have?

**Comment by Rii-** [comment]
Okay thanks ! But forget about the crash, I just tried to reproduce it in Safe Mode and I couldn't make the app crash again, which means that I have a conflicting tweak so no worries, your app works well ! ;)

**Comment by Sidetalker** [comment]
Good to hear! I should have time to add the Settings section and push an update tonight

**Comment by Rii-** [comment]
Awesome ! But it's 11 p.m here so I will give you my feedback in approximately 12 hours ! :)

**Comment by Sidetalker** [comment]
Latest app is here! You should be able to manually choose your localization through Settings. I'm also interested in what that screen shows for your Device Locale

**Comment by Sidetalker** [comment]
FYI, the device locale isn't persisting right now (if you fully quit the app it will go back to the default again) - I'll be fixing that this evening. It also seems to be really crashy right now, it was a bit of a rush job and I'll be cleaning things up.
